### PR TITLE
refactor: restart management & cleanup

### PR DESCRIPTION
**Pull request by swarnimarun**
### Description

Restarts from helper are currently broken as they don't use the initial setup commands.
Also cleanup didn't manage things correctly.

Further we now avoid using the root `/` path, instead we use the `/cedana/scripts` directory for storing scripts and executing them. This ensure we don't overwrite or conflict with any existing files.

Fixes CED-782

### CODE REVIEW

File: cmd/scripts/k8s/cleanup-host.sh [hunk 1]
```diff
@@ -1,23 +1,4 @@
...
-
-systemctl stop cedana.service
-
-
-rm -f /usr/local/bin/cedana
-rm -f /build-start-daemon.sh
-EOT
-
-echo "Clean up completed."
+chroot /host /bin/bash /reset.sh
```
Comment (swarnimarun) [inline cmd/scripts/k8s/cleanup-host.sh:3]:
```suggestion
chroot /host /bin/bash /cedana/scripts/reset.sh
```

File: .github/workflows/pr.yml [hunk 2]
```diff
@@ -136,8 +136,10 @@ jobs:
         run: |
           sudo chmod +x ./cedana
           sudo cp ./cedana /usr/local/bin/
-          sudo cp ./build-start-daemon.sh /build-start-daemon.sh
-          sudo ./setup-host.sh
+          mkdir -p /cedana/scripts
```
Comment (swarnimarun) [inline .github/workflows/pr.yml:139]:
```suggestion
          sudo mkdir -p /cedana/scripts
```

File: pkg/container/utils_linux.go [hunk 3]
```diff
@@ -256,7 +256,7 @@ func Create(root, id string, config *configs.Config) (*RuncContainer, error) {
 	if cm.Exists() {
 		// pids, err := cm.GetAllPids()
 		// Reading PIDs can race with cgroups removal, so ignore ENOENT and ENODEV.
-		if err != nil && !errors.Is(err, os.ErrNotExist) && !errors.Is(err, unix.ENODEV) {
+		if !errors.Is(err, os.ErrNotExist) && !errors.Is(err, unix.ENODEV) {
```
Comment (yashanand1910) [inline pkg/container/utils_linux.go:259]: This will always return err.

**Review by yashanand1910 — CHANGES_REQUESTED** [review]
Check comments - cause of regression failure.

File: pkg/container/utils_linux.go (hunk 3 above)
Comment (swarnimarun) [reply pkg/container/utils_linux.go:259]: I think this was a golint suggestion... reverting let's see.

Comment (swarnimarun) [inline pkg/container/utils_linux.go:259]:
```suggestion
		if err != nil && !errors.Is(err, os.ErrNotExist) && !errors.Is(err, unix.ENODEV) {
```

File: cmd/scripts/k8s/setup-host.sh [hunk 4]
```diff
@@ -4,20 +4,27 @@ chroot /host /bin/bash <<'EOT'
...
+    echo -n > /var/log/cedana-daemon.log
+else
+    echo "No cedana-daemon.log file found"
+    echo "Skipping $APP_NAME service stop"
+fi
 EOT
 
+# Create Cedana directories
+mkdir -p /host/cedana /host/cedana/bin /host/cedana/scripts
+
```
Comment (swarnimarun) [inline cmd/scripts/k8s/setup-host.sh:21]:
```suggestion

# We load the binary from docker image for the container
```

**Review by nravic — APPROVED** [review]
lgtm

**Pull request merged**